crates/typst-macros/src/elem.rs

```diff
@@ -214,228 +219,215 @@ fn create(element: &Elem) -> TokenStream {
             #(#push_field_methods)*
             #(#field_style_methods)*
 
-            /// The element's span.
-            pub fn span(&self) -> ::typst::syntax::Span {
-                self.0.span()
+            /// Set the element's span.
+            pub fn spanned(mut self, span: ::typst::syntax::Span) -> Self {
```

Comment:
any reason these three methods (spanned/located/labelled) are inherent instead of being part of `NativeElement`? if it's because of object safety, we can do `where Self: Sized` again.